src/webui/www/private/scripts/contextmenu.js

```diff
@@ -413,6 +433,43 @@ var TorrentsTableContextMenu = new Class({
             }
             categoryList.appendChild(el);
         });
+    },
+
+    updateTagsSubMenu: function(tagList) {
+        const contextTagList = $('contextTagList');
+        while (contextTagList.firstChild)
+            contextTagList.removeChild(contextTagList.firstChild);
+
+        contextTagList.appendChild(new Element('li', {
+            html: '<a href="javascript:torrentAddTagsFN();">'
+                + '<img src="images/qbt-theme/list-add.svg" alt="QBT_TR(Add...)QBT_TR[CONTEXT=TransferListWidget]"/>'
+                + ' QBT_TR(Add...)QBT_TR[CONTEXT=TransferListWidget]'
+                + '</a>'
+        }));
+        contextTagList.appendChild(new Element('li', {
+            html: '<a href="javascript:torrentRemoveAllTagsFN();">'
+                + '<img src="images/qbt-theme/edit-clear.svg" alt="QBT_TR(Remove All)QBT_TR[CONTEXT=TransferListWidget]"/>'
+                + ' QBT_TR(Remove All)QBT_TR[CONTEXT=TransferListWidget]'
+                + '</a>'
+        }));
+
+        const sortedTags = [];
+        for (let key in tagList)
+            sortedTags.push(tagList[key].name);
+        sortedTags.sort();
+
+        for (let i = 0; i < sortedTags.length; ++i) {
+            const tagName = sortedTags[i];
+            const tagHash = genHash(tagName);
+            const el = new Element('li', {
+                html: '<a href="#Tag/' + tagHash + '" onclick="event.preventDefault(); torrentSetTagsFN(\'' + tagHash + '\', !event.currentTarget.getElement(\'input[type=checkbox]\').checked);">'
+                    + '<input type="checkbox" onclick="this.checked = !this.checked;"> ' + escapeHtml(tagName)
+                    + '</a>'
+            });
+            if (i === 0)
```

Comment:
Can this check be moved outside the for loop? Seems we can call it if `sortedTags.length > 0`.

Reply:
Then we'll have to do even more work to find first tag element once more.

Reply:
You're right, I overlooked the part where we create the element in the line directly above. You can ignore this.